operators/pkg/controller/elasticsearch/driver/downscale.go

```diff
@@ -67,6 +75,19 @@ func HandleDownscale(
 	return results
 }
 
+// noOnGoingDeletion returns true if some pods deletion or creation may still be in progress
+func noOnGoingDeletion(downscaleCtx downscaleContext, actualStatefulSets sset.StatefulSetList) (bool, error) {
+	// Pods we have may not match replicas specified in the StatefulSets spec.
+	// This can happen if, for example, replicas were recently downscaled to remove a node,
+	// but the node isn't completely terminated yet, and may still be part of the cluster.
+	// Moving on with downscaling more nodes may lead to complications when dealing with
+	// Elasticsearch shards allocation excludes (should not be cleared if the ghost node isn't removed yet)
+	// or zen settings (must consider terminating masters that are still there).
+	// Let's retry once expected pods are there.
+	// MatchActualPods also matches any pod not created yet, for which we'll also requeue.
+	return actualStatefulSets.MatchActualPods(downscaleCtx.k8sClient, downscaleCtx.es)
+}
```

Comment:
I'm in favor of small functions but here I wonder if it's worth it to have `noOnGoingDeletion` and `canClearVotingConfigExclusions` that just call `actualStatefulSets.MatchActualPods` and bring a little additionnal indirection to follow the flow of the execution?

Reply:
My reasoning was that the big part of this function is actually the comment block that comes along with it. Including it in `HandleDownscale` kind of breaks the flow of reading `HandleDownscale`, which I see as the high-level flow of downscales.
I think `HandleDownscale` is nice to read as just "is there any ongoing deletion? No? ok continue". And if you want the details you get into the function implementation, read the comment to understand it's tricky, and realize it's just a call to `MatchActualPods` so we're lucky it's that simple.
WDYT?